### PR TITLE
feat: ✨sd-radio & sd-radio-group 

### PR DESCRIPTION
**Pull request by yoezlem**

### CODE REVIEW

File: packages/components/src/components/radio-group/radio-group.ts [hunk 1]
```diff
@@ -27,20 +31,23 @@ import type SdRadioButton from '../radio-button/radio-button';
...
+ * @csspart form-control - The form control that wraps the label, input, and error text.
  * @csspart form-control-label - The label's wrapper.
  * @csspart form-control-input - The input's wrapper.
- * @csspart form-control-help-text - The help text's wrapper.
- * @csspart button-group - The button group that wraps radio buttons.
- * @csspart button-group__base - The button group's `base` part.
+ * @csspart form-control-error-text - The error text's wrapper.
  */
+
+// TODO: integrate these parts into the component when radio-button is converted to a solid component
```
Comment (karlbaumhauer) [inline packages/components/src/components/radio-group/radio-group.ts:42]: pls add the respective GitHub Issue for the todo

**Review by karlbaumhauer — CHANGES_REQUESTED** [review]

**Comment by yoezlem** [comment]
> @yoezlem did we align all the naming with design? As we had the button/radio-button/radio-group in many cases and I have the feeling that it is still not 100% aligned, I suggest to go through it once again before merging. So that we dont have any breaking changes afterwards.

Yes, we have the <sd-radio>, which is a single radio, and the <sd-radio-group>. For button-styled elements, we also have the <sd-radio-button-group>. This should align with the design.

**Review by karlbaumhauer — APPROVED** [review]

**Review by azraefendic — APPROVED** [review]

**Comment by mariohamann** [comment]
CleanShot 2023-10-18 at 10 06 22@2x

Could you improve the slots here?

File: packages/components/src/components/radio/radio.ts [hunk 2]
```diff
@@ -142,67 +167,14 @@ export default class SdRadio extends SolidElement {
...
-        font-size: var(--sd-input-font-size-large);
-      }
-
-      .radio__checked-icon {
-        display: inline-flex;
-        width: var(--toggle-size);
-        height: var(--toggle-size);
-      }
-
       .radio__control {
```
Comment (mariohamann) [inline packages/components/src/components/radio/radio.ts:170]: Should be replaceable with `flex-initial shrink-0`

File: packages/components/src/components/radio/radio.ts [hunk 3]
```diff
@@ -142,67 +167,14 @@ export default class SdRadio extends SolidElement {
...
-      }
-
-      /* Checked + hover */
-      .radio.radio--checked:not(.radio--disabled) .radio__control:hover {
-        border-color: var(--sd-color-primary-500);
-        background-color: var(--sd-color-primary-500);
       }
 
       /* Checked + focus */
       :host(:focus-visible) .radio__control {
```
Comment (mariohamann) [inline packages/components/src/components/radio/radio.ts:175]: Please remove `.radio__control` and refer to the part it should select.

**Review by mariohamann — APPROVED** [review]

**Comment by coraliefeil** [comment]
Left comments on chromatic ... pls reassign for another review. Thank you <3

**Comment by coraliefeil** [comment]
Review done. Left comments on Chromatic. Pls reassign when ready. Thanks <3

**Comment by coraliefeil** [comment]
Approved. Thanks a lot.

File: packages/components/src/components/radio/radio.ts [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+            /* sizes, fonts */
+            sm: 'small-size text-sm',
+            lg: 'text-base'
+          }[this.size]
+        )}
+      >
+        <span
+          part="${`${this.checked ? 'control--checked' : 'control--unchecked'}`}"
+          class=${cx(
+            'radio__control flex-initial shrink-0 relative inline-flex items-center justify-center border rounded-full bg-white h-4 w-4',
```
Comment (mariohamann) [inline packages/components/src/components/radio/radio.ts:121]: Please remove `radio__control`

File: packages/components/src/components/radio-group/radio-group.ts [hunk 5]
```diff
@@ -290,32 +403,83 @@ export default class SdRadioGroup extends SolidElement implements SolidFormContr
...
+      }
+
+      .form-control-input--horizontal ::slotted(sd-radio:last-of-type) {
+        margin-right: 0;
+      }
+
+      /* Label */
+      .form-control--has-label .form-control__label {
+        display: flex;
+      }
```
Comment (mariohamann) [inline packages/components/src/components/radio-group/radio-group.ts:463]: Should easily be doable with `cx(hasError: "flex")` or similar.

Comment (mariohamann) [reply packages/components/src/components/radio-group/radio-group.ts:463]: And remove the class above then.

File: packages/components/src/components/radio-group/radio-group.ts [hunk 6]
```diff
@@ -290,32 +403,83 @@ export default class SdRadioGroup extends SolidElement implements SolidFormContr
...
+      .form-control--has-error-text .form-control__error-text {
+        display: flex;
+      }
+
+      .visually-hidden {
+        width: 1px;
+        height: 1px;
+        margin: -1px;
+        clip: rect(0, 0, 0, 0);
+      }
```
Comment (mariohamann) [inline packages/components/src/components/radio-group/radio-group.ts:480]: Please use Taiwliind class `sr-only`.

File: packages/components/src/components/radio-group/radio-group.ts [hunk 7]
```diff
@@ -290,32 +403,83 @@ export default class SdRadioGroup extends SolidElement implements SolidFormContr
...
+
+      :host([required]) .form-control--has-label .form-control__label::after {
+        content: '*';
+        margin-left: 2px;
+      }
+
+      /* error text */
+      .form-control--has-error-text .form-control__error-text {
+        display: flex;
+      }
```
Comment (mariohamann) [inline packages/components/src/components/radio-group/radio-group.ts:473]: Should easily be doable with cx(hasError...) or similar.

**Review by mariohamann — CHANGES_REQUESTED** [review]

File: packages/components/src/components/radio-group/radio-group.ts (hunk 5 above)
Comment (yoezlem) [reply packages/components/src/components/radio-group/radio-group.ts:463]: Good point. Could do this also with the label class! thx

File: packages/components/src/components/radio-group/radio-group.ts (hunk 7 above)
Comment (yoezlem) [reply packages/components/src/components/radio-group/radio-group.ts]: Had an issue with the hidden tailwind class, which is always "!important" -> didn't know that.
But could solve it with a statement. 

So, CSS class is now removed! 🙂

**Comment by mariohamann** [comment]
Removed myself – if you need a re-review, feel free to add me.

**Review by Vahid1919 — APPROVED** [review]

**Review by mariohamann — APPROVED** [review]

**Pull request merged**